x-pack/plugins/security_solution/public/management/pages/endpoint_hosts/view/details/index.tsx

```diff
@@ -98,23 +82,40 @@ export const EndpointDetailsFlyout = memo(() => {
     []
   );
 
-  const tabs = [
-    {
-      id: EndpointDetailsTabsTypes.overview,
-      name: i18.OVERVIEW,
-      content:
-        hostDetails === undefined ? (
-          ContentLoadingMarkup
-        ) : (
-          <EndpointDetails details={hostDetails} policyInfo={policyInfo} hostStatus={hostStatus} />
-        ),
-    },
-    {
-      id: EndpointDetailsTabsTypes.activityLog,
-      name: i18.ACTIVITY_LOG.tabTitle,
-      content: <EndpointActivityLog activityLog={activityLog} />,
-    },
-  ];
+  const getTabs = useCallback(
```

Comment:
this feels like it should be a `useMemo()` instead that uses the `hostDetails.agent.id` to build up the array.

Reply:
So I did want to use that, but at this point, we `hostDetails` could be `undefined` and consequently, we can't assign `undefined` to `selected_id`.

Reply:
you could do that check inside of the `useMemo()` and just return an empty array if the id is undefined. Nothing that should hold this PR up, so 🚢 😄